app/src/main/java/com/conupods/OutdoorMaps/View/MapsActivity.java

```diff
@@ -0,0 +1,318 @@
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+
+import static android.view.inputmethod.EditorInfo.IME_ACTION_DONE;
+import static android.view.inputmethod.EditorInfo.IME_ACTION_SEARCH;
+
+public class MapsActivity extends FragmentActivity implements OnMapReadyCallback {
+
+    private static final String TAG = "MapsActivity";
+    private static final int ERROOR_DIALOG_REQUEST = 9001;
+    private static final float DEFAULT_ZOOM = 15f;
+
+
+    private GoogleMap mMap;
+
+    private final String COURSE_LOCATION_PERMISSION = Manifest.permission.ACCESS_COARSE_LOCATION;
+    private final String FINE_LOCATION_PERMISSION = Manifest.permission.ACCESS_FINE_LOCATION;
+
+    //Activity Components
+    //private  Button locationBtn;
+    private View mapView;
+    private EditText searchBar;
+
+
+    //Variables for logic
+    Boolean permissionsGranted = false;
+    private static final int LOCATION_PERMISSION_REQUEST_CODE = 1234;
+    private static final int RESOLVABLE_API_ERROR_REQUEST_CODE = 51;
+    Location lastKnownLocation;
+    LocationCallback locationCallback;
+
+    //Providers
+    //TODO Might need to update to more recent version
+    private FusedLocationProviderClient fusedLocationProvider;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_maps);
+        initializeMap();
+
+        if(!permissionsGranted)
+            getLocationPermission();
+        else{
+            getDeviceCurrentLocation();
+        }
+        
+    }
+
+
+    private void initializeMap() {
+        Log.d(TAG, "Initializing Map...");
+
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
+                .findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this);
+
+        searchBar = (EditText) findViewById(R.id.search);
+        //TODO Remove to create custom current location button
+        searchBar.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView textView, int actionId, KeyEvent keyEvent) {
+                if(actionId == IME_ACTION_SEARCH
+                || actionId == IME_ACTION_DONE
+                || keyEvent.getAction() == KeyEvent.ACTION_DOWN
+                || keyEvent.getAction() == keyEvent.KEYCODE_ENTER
+                )
+                {
+                   //TODO Logic for searching goes here
+                   geoLocate();
+                }{
+
+                }
+                return false;
+            }
+        });
+
+
+        mapView = mapFragment.getView();
+    }
+
+    private void geoLocate(){
+        //DO SOMETHINF
```

Comment:
typo, also is this not part of ur feature to locate urself?

Reply:
Removed it, kept the TODO comment though